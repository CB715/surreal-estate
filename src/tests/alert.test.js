import React from "react";
import { render } from "@testing-library/react";

import Alert from "../components/alert";

describe("alert", () => {
  it("displays the alert error message", () => {
    const alert = render(<Alert message="Error!" />);
    const errorMessage = alert.getByText("Error!");

    expect(errorMessage.textContent).toBe("Error!");
  });
  it("displays the alert success message", () => {
    const alert = render(<Alert message="Success!" success />);
    const errorMessage = alert.getByText("Success!");

    expect(errorMessage.textContent).toBe("Success!");
  });
  it("does not render a message if props is empty", () => {
    const { asFragment } = render(<Alert message="" />);
    const alert = asFragment();

    expect(alert).toMatchSnapshot();
  });
});
