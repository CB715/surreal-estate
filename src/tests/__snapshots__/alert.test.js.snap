// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alert does not render a message if props is empty 1`] = `<DocumentFragment />`;
